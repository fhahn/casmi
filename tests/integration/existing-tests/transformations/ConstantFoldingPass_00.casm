// transform, cmdline "--cf"

CASM ConstantFoldingPass

init foo

function bar : -> Int

rule foo = 
{
    bar := 2 * 3 + 4
    
    program(self) := undef
}

