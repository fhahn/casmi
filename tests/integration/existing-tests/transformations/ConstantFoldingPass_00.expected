foo=STATEMENTS(LET CONST(ident c0__,num 10),PARBLOCK(STATEMENTS(UPDATE(ident bar,ident c0__),STATEMENTS(UPDATE(LOOKUP(ident program,LIST(SELF,null)),UNDEF),null)),null))
